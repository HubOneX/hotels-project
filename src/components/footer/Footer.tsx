import styled from "@emotion/styled";
import { Link, Typography } from "@mui/material";
import React, { FC } from "react";
import colors from "../../constants/colors";
import { FlexCenterAlignedBox } from "../utilityComponents/styledComponents";

// Added so the spaces around the Link are being rendered in a way that doesn't clutter the code,
// normally I would use translations with these spaces already in them.
const madeBy = "Made by ";
const forInterview = " for a technical interview.";

const FooterBox = styled(FlexCenterAlignedBox)({
  height: "10vh",
  width: "100%",
  backgroundColor: colors.blue,
  marginTop: "50px",
  boxShadow: `0px 6px 30px 0px ${colors.midGrey}`,
  color: colors.white,
});

const StyledLink = styled(Link)({
  color: colors.white,
  fontWeight: 500,
  letterSpacing: "1px",
});

type Props = {
  fixedAtBottom: boolean;
};

const Footer: FC<Props> = ({ fixedAtBottom }) => {
  return (
    <FooterBox
      sx={{
        position: `${fixedAtBottom ? "fixed" : "inherit"}`,
        bottom: `${fixedAtBottom ? "0" : "inherit"}`,
      }}
    >
      <Typography sx={{ marginLeft: "30px" }}>
        {madeBy}
        <StyledLink
          target="_blank"
          href="https://github.com/HubOneX"
          underline="hover"
        >
          Hubert Dardzinski
        </StyledLink>
        {forInterview}
      </Typography>
    </FooterBox>
  );
};

export default Footer;
