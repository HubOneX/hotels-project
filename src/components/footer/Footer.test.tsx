import renderer from "react-test-renderer";
import Footer from "./Footer";

describe("Footer", () => {
  it("should match the snapshot", () => {
    const container = renderer.create(<Footer fixedAtBottom={false} />);

    expect(container).toMatchSnapshot();
  });
});
