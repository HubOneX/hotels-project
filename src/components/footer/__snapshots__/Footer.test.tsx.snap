// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match the snapshot 1`] = `
<div
  className="MuiBox-root css-1x3cxik"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-19k4ldb-MuiTypography-root"
  >
    Made by 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1mz76tb-MuiTypography-root-MuiLink-root"
      href="https://github.com/HubOneX"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      Hubert Dardzinski
    </a>
     for a technical interview.
  </p>
</div>
`;
